platforms/software/dependency-management/src/integTest/groovy/org/gradle/integtests/internal/component/resolution/failure/ResolutionFailureHandlerIntegrationTest.groovy

```diff
@@ -306,6 +304,9 @@ class ResolutionFailureHandlerIntegrationTest extends AbstractIntegrationSpec {
         verifyAll(receivedProblem(0)) {
             fqid == 'dependency-variant-resolution:incompatible-multiple-nodes'
         }
+        verifyAll(receivedProblem(1)) {
```

Comment:
🤔 How come we get an additional problem reported here?

Reply:
The single incompatibility failure reported to the failure handler is ultimately packaged by dependency management as a multi-cause exception caused by 2 `ModuleVersionResolveException`s (representing the incompatibility in both directions).  

The exception deduplication code knows to only report this only once when it summarizes the failure, but deduplication is not yet implemented for Problems.  Once the IDE team adds that, the expectation of 2 problems here will show up as a failed test and they'll see to remove it.  Until then, this is just how it works - there's no equality defined on the exception types used here.